cloud/pkg/cloudhub/servers/httpserver/secretsutil.go

```diff
@@ -98,3 +105,21 @@ func CreateCloudCoreSecret(certDER, key []byte) error {
 	}
 	return CreateSecret(cloudCoreCert, NamespaceSystem)
 }
+
+func CreateNamespaceIfNeeded(cli *kubernetes.Clientset, ns string) error {
+	c := cli.CoreV1()
+	if _, err := c.Namespaces().Get(ns, metav1.GetOptions{}); err == nil {
+		return nil
+	}
+	newNs := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ns,
+			Namespace: "",
+		},
+	}
+	_, err := c.Namespaces().Create(newNs)
+	if err != nil && errors.IsAlreadyExists(err) {
+		err = nil
+	}
+	return err
```

Comment:
Please refer to: https://github.com/kubeedge/kubeedge/blob/master/keadm/cmd/keadm/app/cmd/util/k8sinstaller.go#L73

Reply:
I looked at these two functions and they are the same, the function CreateNamespaceIfNeeded is from k8s.

Reply:
just keep it here.